### ISSUE TITLE
Remove `IS_PERSONAL_SERVER` config in favor of specific flags per feature

### ISSUE THREAD

**Issue by brassy-endomorph** [scidsg/hushline#646]
**Is your feature request related to a problem? Please describe.**

The notion of a "personal server" is ambiguous and internally on the Hushline team is used to differentiate between the flagship instance at `tips.hushline.app` and one someone else hosts. The flag `IS_PERSONAL_SERVER` is set to change the behavior of the app.

Now that we have Stripe integration, there are open tickets and discussions that require features to be conditionally gated on payment, and there are additionally open discussions that blend both paid features with "personal server." This is confusing and adds complexity to the logic of the app.

**Describe the solution you'd like**

1. Any self-hosted instance should have all features enabled for all users by default without gating on a paid tier
2. Completely remove the notion of "personal server" from the app as there should be no distinction of any sort between the flagship ("ours") and a hosted instance ("theirs"). 
3. For any feature that can be conditionally enabled, have a single feature flag to enable or disable that. These may be boolean or if they require payment could be ternary with values being `enabled`/`disabled`/`paid-only`.

**Additional Context**

Generally speaking, one flag should enable or disable a single feature so that its meaning is totally clear and transparent. Overuse of multi-purpose flags like this can make it hard to reason about what flipping a single flag does. Further, it hides dependencies between different features by lumping them together. If we want to decouple them later, it's far more work rather than explicitly surfacing the complexity at the time a feature is implemented.

That said, it simplifies testing as we only have to test two types of servers (hosted and personal) instead of matrix testing many combinations of configurations.

**Comment by brassy-endomorph** [comment]
Now that I've looked into this more closely, there's only a few places where the code cares about the `is_personal_server` flag, and this is mostly templates to change what's displayed.

This logic for the index page be changed to depend on `single_user_mode` which is a var calculated on `SELECT count(*) FROM users` being `> 1`. But if we want a single user mode, that should prevent the creation of multiple users so that messages aren't delivered to inboxes that don't exist. This might need to impact the `/to/<user>` route as well so that it always redirects. Maybe single user mode is a flag via env var that disables a bunch of things in the UI and the backend.

The Tor banner could be done by detecting if the user agent is Tor. I know SecureDrop does this somehow, but I can't recall the logic. It doesn't require an `is_personal_server` flag, and we don't want to show it to Tor users on the flagship instance for example.

Not sure why `/vision` is turned off for `is_personal_server`. What's the logic there?

It's also unclear why `is_personal_server` impacts if users can use SMTP forwarding.

So far as I can tell, these are all independent configs that are bundled under `is_personal_server`, and they should be managed separately.

**Comment by micahflee** [comment]
How would the feature flags work? Would it just be environment variables like:

- `FEATURE_PAID` for supporting paid features (only used on the flagship instance)
- `FEATURE_SINGLE_USER` for a single-user instance
- etc.

Also if I remember correctly, the SMTP forwarding thing was just about a string in the template that says "✊ Email forwarding is powered by Riseup.net." This is true on the flagship service, but if someone is hosting their own service and they set their own global SMTP settings, it shouldn't claim to users that it's powered by Riseup. So maybe for things like this there should be a `FEATURE_FLAGSHIP` var, meaning only run this code on the flagship instance?

**Comment by brassy-endomorph** [comment]
> FYI Tor Browser's user agent string makes it look like Windows in Firefox. SecureDrop is Tor-only, so it just always shows it.

I tracked it down and it was for detecting tor2web, so yeah, there's not an easy way to do this other than looking up the IP against the known exit notes.

**Comment by brassy-endomorph** [comment]
> `FEATURE_PAID` for supporting paid features (only used on the flagship instance)

I think this is actually a separate config. We have a flag like `PAYMENTS_ENABLED` which is a boolean and then it looks for Stripe keys. If it's not enabled, then all paid features are free (or their default whatever that is)

So the combo to enable Stripe payments for Aliases would be something like:

- `PAYMENTS_ENABLED` to turn on payment features
- `STRIPE_*` for the actual configs for Stripe
- `USER_ALIASES_MODE` to allow users to create aliases with values: `always`, `premium`, `never`

Because maybe later someone wants to host and is like "fuck it, everyone can have aliases, but I want to charge for file storage." This would mean payment is something that is turned on and off independently of the effects it has on other things. We might have to 503 on some routes if there's bad configuration like if `PAYMENTS_ENABLED=false` and `USER_ALIASES_MODE=premium` but if that's documented, it's fine. This would also mean the app can mostly work on bad config and only impacted routes/features would "break" on misconfiguration. Better than throwing an exception on boot and never getting the app off the ground for a bad config IMO.

> Also if I remember correctly, the SMTP forwarding thing was just about a string in the template that says

So to me this string is actually a config value like `SMTP_FORWARDING_MESSAGE` and if there's no value set, nothing appears when SMTP forwarding is enabled.

---

Like IMO we shouldn't have a notion of `FLAGSHIP` in the app code, and maybe the way we retain that mental model is to have multiple .env files or multiple compose files that simulate different environments. Maybe even a `flagship` compose file (and a rPi one too for that mode) so the basic testing/dev can be done quite easily, but having a `FLAGSHIP` feature flag just feels like inverting the idea of `IS_PERSONAL_SERVER` and we're back in the same position.

When I worked at `$COMPANY` and we had a large service we hosted and did on-prem, so far as I ever saw, we never had flags like `FLAGSHIP`, and everything was fine-grained configs for single features so that everything could be toggled and adjusted individually. We had... so many configs and features. But never a `FLAGSHIP` one. And while maybe some places do that, I very much did not get the impression that's a best practice.
